[DOCS] Adds autoscaling info to Anomaly detection at scale (#1659)
elastic/stack-docs: docs/en/stack/ml/anomaly-detection/anomaly-detection-scale.asciidoc

[discrete]
[[node-sizing]]

> == 1. Consider node sizing/autoscaling and configuration

Reviewer: IMO these terms should all be "ands" instead of an "and" and an "or":
```suggestion
== 1. Consider autoscaling, node sizing, and configuration
```

as fault tolerance. If running many jobs, even small memory ones, then consider 
increasing the number of nodes in your environment.

> In {ecloud}, you can define policies that scales up or down your {ml} nodes 

Reviewer: ```suggestion
In {ecloud}, you can define policies that scale your {ml} nodes up or down
```

based on node requirements by using {ref}/xpack-autoscaling.html[autoscaling]. 
Set `xpack.ml.max_ml_node_size` to define the maximum possible size of a {ml} 

> node. When you try to create a {anomaly-job} that is larger than the maximum 

Reviewer: ```suggestion
node. When you try to create an {anomaly-job} that is larger than the maximum 
```

size a node can support, it instantly fails as autoscaling cannot add a node big 
enough to run the job. On a self-managed deployment, make sure that 
`xpack.ml.max_model_memory_limit` is set according to the available resources of 
the {ml} node. In this case, if you have a job with a larger model, it instantly 

> fails otherwise it cannot be opened due to lack of resources and it remains in 

Reviewer: IMO we need a semi-colon or period here:
```suggestion
fails. Otherwise, it cannot be opened due to lack of resources and it remains in 
```

> In {ecloud}, you can define policies that scale your {ml} nodes up or down
> based on node requirements by using {ref}/xpack-autoscaling.html[autoscaling]. 
> Set `xpack.ml.max_ml_node_size` to define the maximum possible size of a {ml} 
> node. If you try to create an {anomaly-job} that is larger than the maximum 
> size a node can support, it instantly fails as autoscaling cannot add a node big 
> enough to run the job. On a self-managed deployment, make sure that 
> `xpack.ml.max_model_memory_limit` is set according to the available resources of 
> the {ml} node. In this case, if you have a job with a larger model, it instantly 
> fails. Otherwise, it cannot be opened due to lack of resources and it remains in 
> the `opening` state.

Reviewer 2: We shouldn't recommend setting `xpack.ml.max_ml_node_size` in ESS, as the ESS infrastructure sets it to the correct value. Also, it's not compulsory to set `xpack.ml.max_model_memory_limit` in a self-managed cluster. The UI will still never suggest a model memory limit that's too high to be opened in the cluster even if this setting is not set. That setting was primarily added as a way to protect Cloud nodes against people creating jobs that were too big, but Cloud already sets this setting itself. Saying "make sure" makes it sound like this is something that's essential for people to do when it's really not in most cases.

```suggestion
In {ecloud}, you can enable {ref}/xpack-autoscaling.html[autoscaling] so that the
{ml} nodes in your cluster will scale up or down based on current {ml} memory
requirements. The {ecloud} infrastructure will allow you to create {ml-jobs} up
to the size that would fit on the maximum node size that the cluster can scale to
(usually somewhere between 58GB and 64GB) rather than what would fit in the
current cluster. If you are attempting to use autoscaling outside of {ecloud} then
set `xpack.ml.max_ml_node_size` to define the maximum possible size of a {ml} 
node. Creating {ml-jobs} with model memory limits larger than the maximum the
maximum node size can support will not be allowed, as autoscaling cannot add
a node big enough to run the job. On a self-managed deployment, you can set 
`xpack.ml.max_model_memory_limit` according to the available resources of 
the {ml} node. This will stop you creating jobs with model memory limits too high
to open in your cluster.
```
